crates/astria-conductor/src/conductor/inner.rs

```diff
@@ -244,13 +272,50 @@ impl Conductor {
             info!("all tasks shut down regularly");
         }
         info!("shutting down");
+
+        restart_or_shutdown
     }
 }
 
 #[instrument(skip_all)]
-fn report_exit(exit_reason: eyre::Result<&str>, message: &str) {
+fn report_exit(exit_reason: &ExitReason, message: &str) {
     match exit_reason {
-        Ok(reason) => info!(%reason, message),
-        Err(reason) => error!(%reason, message),
+        ExitReason::ShutdownSignal => info!("received shutdown signal, {}", message),
+        ExitReason::TaskFailed {
+            name: _,
+            error,
+        } => error!(%error, message),
+    }
+}
```

Comment:
Can you bring back the previous structure?

```suggestion
        ExitReason::ShutdownSignal => info!(reason = "received shutdown signal", message),
        ExitReason::TaskFailed {
            name: task,
            error: reason,
        } => error!(%reason, %task, message),
    }
}
```